src/app/modules/registration/pages/obs-location/obs-location.page.ts

```diff
@@ -44,35 +46,43 @@ export class ObsLocationPage implements OnInit, OnDestroy {
 
   async ngOnInit() {
     const id = this.activatedRoute.snapshot.params['id'];
+    const geoHazard = this.activatedRoute.snapshot.params['geoHazard'];
+    const lat = this.activatedRoute.snapshot.queryParams['lat'];
+    const lon = this.activatedRoute.snapshot.queryParams['lon'];
+    const locationId = this.activatedRoute.snapshot.queryParams['locationId'];
     if (id) {
       // Edit an existing draft
       this.draft = await this.draftService.load(id);
       this.geoHazard = this.draft.registration.GeoHazardTID;
-    } else if (this.activatedRoute.snapshot.params['geoHazard']) {
+    } else if (geoHazard) {
       // New draft - will be created later
-      this.geoHazard = parseInt(this.activatedRoute.snapshot.params['geoHazard'], 10);
+      if (isNaN(geoHazard)){
+        const geoHazardCapLetter = geoHazard.charAt(0).toUpperCase() + geoHazard.slice(1);
+        this.geoHazard = parseInt(GeoHazard[geoHazardCapLetter], 10);
+      } else {
+        this.geoHazard = parseInt(geoHazard, 10);
+      }
     }
     if (this.geoHazard == null) {
       // No geohazard found, use app mode
       const userSettings = await firstValueFrom(this.userSettingService.userSetting$);
       this.geoHazard = userSettings.currentGeoHazard[0];
     }
+    if(lat && lon && locationId == undefined){
```

Comment:
Det holder å sjekke `if (lat && lon)` her.

Reply:
hmm poenget er at hvis man skriver alle lat, lon og locationId (som er feil, man burde velge enten lat. lon set eller locationId) så vise obspage enten lokasjonen med definert lat lon og ignorer locationId (eller omvendt basert på hva som står i den første ifen). Jeg tenkte at hvis man skriver alt så burde vi kanksje returnere en default lokasjon, siden urlen er feilete? eller vi kan prioritere locationId. feilen vil skje bare hvis man skriver urlen manuelt for iskart.no vil alltid velge en ritkig url, men men